odoo/fields.py

```diff
@@ -1583,6 +1586,12 @@ def translate(record_id, value):
 
         return translate
 
+    def convert_to_cache(self, value, record, validate=True):
+        value = super().convert_to_cache(value, record, validate)
+        if validate and value and self.pattern and not any(re.findall(self.pattern[1], value)):
```

Comment:
`any(re.findall(self.pattern[1], value))` could be simplified to `re.search(self.pattern[1], value)`
Shouldn't we check that `value` matches the regexp as a whole?
Moreover, no flags are passed to pattern match function.  What about multi-line text fields?

Reply:
match vs search is just an opinion, when you'll write your pattern you'll add `.*(...).*` or `^(...)$` depending on the choice made. IMO any of the two is fine.